src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -296,9 +299,12 @@ private static T NextOrDefault<T>(IEnumerator<T> enumerator)
             /// <summary>
             /// Return all the spans that appear in only one of "latestSpans" or "previousSpans".
             /// </summary>
-            private static IEnumerable<SnapshotSpan> Difference<T>(IEnumerable<ITagSpan<T>> latestSpans, IEnumerable<ITagSpan<T>> previousSpans, IEqualityComparer<T> comparer)
+            private static DiffResult Difference<T>(IEnumerable<ITagSpan<T>> latestSpans, IEnumerable<ITagSpan<T>> previousSpans, IEqualityComparer<T> comparer)
                 where T : ITag
             {
+                List<SnapshotSpan> added = null;
```

Comment:
much simpler to just 'new' these here.  You could even pool them if you really wanted to since the DiffResult is going to make a copy of them in the constructor.